### PR TITLE
PR for post route fixes and meter_api file name

### PR DESCRIPTION
**Pull request by avpeery**
This PR request completes the following fixes: changing API meter file name to meter.py, accepts both trailing and non-trailing routes, and accepts json object vs previous form.

Continue WIP:
-Minor error handling on POST route to ensure meter_id does not already exist
-Finish put route

### CODE REVIEW

File: web/web/api/v1/meter.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from sqlalchemy.orm.exc import NoResultFound
 from sqlalchemy.orm.exc import MultipleResultsFound
 
+#enables trailing and non-trailing slash routes
+app.url_map.strict_slashes = False
```
Comment (jongoncalves) [inline web/web/api/v1/meter.py:14]: this should move to the `web/__init__.py` (where `app` gets instantiated) so we don't have to keep setting it in every api endpoint file.

Actually we'll end up moving it to a config file, but that's for future us.

File: web/web/api/v1/meter.py [hunk 2]
```diff
@@ -84,15 +86,17 @@ def get_meter_schema():
...
-#     try:  
-#        # meter = Meter.query.filter_by(meter_id=meter_id).one()
-#         req = request.get_json()
-#         return jsonify(req)
+#     try:
+#         #checks if meter exists
+#         Meter.query.filter_by(meter_id = meter_id).one()
+#         new_meter_data = request.get_json()
 
+#         return jsonify(req)
```
Comment (jongoncalves) [inline web/web/api/v1/meter.py:97]: Can we get rid of this commented out code?

File: web/web/api/v1/meter.py [hunk 3]
```diff
@@ -102,25 +106,40 @@ def add_meter_info():
...
-    is_archived = request.form.get("is_archived").upper() == "TRUE"
+    new_meter = request.get_json()
+
+    meter_id = new_meter['meter_id']
+    utility_id = new_meter['utility_id']
+    service_location_id = new_meter['service_location_id']
+    feeder = new_meter['feeder']
+    substation = new_meter['substation']
+    meter_type = new_meter['meter_type']
+    is_active = new_meter['is_active'].upper() == "TRUE"
```
Comment (jongoncalves) [inline web/web/api/v1/meter.py:117]: If I don't pass in `is_active` in the body of the request, will this blow up?

File: web/web/api/v1/meter.py [hunk 4]
```diff
@@ -102,25 +106,40 @@ def add_meter_info():
...
         db.session.add(meter)
         db.session.commit()
+        
+        return jsonify({'success': meter.meter_id})
 
     except (MultipleResultsFound,NoResultFound) as e:
-        #no results or multiple results found 
+        #no results or multiple results found for service location
         print(e)
-        return 'error'
```
Comment (jongoncalves) [inline web/web/api/v1/meter.py:123]: Let's return an actual json error response.

We'll eventually write a response wrapper object, but for now, something like:
```python
return jsonify({'error': 'my error was....'})
```
should work

File: web/web/api/v1/meter.py [hunk 5]
```diff
@@ -102,25 +106,40 @@ def add_meter_info():
...
 
 # if __name__ == '__main__':
 #     connect_to_db(app)
```
Comment (jongoncalves) [inline web/web/api/v1/meter.py:145]: can we also remove this commented out code?

File: web/web/api/v1/meter.py (hunk 1 above)
Comment (avpeery) [reply web/web/api/v1/meter.py:14]: Ok! will do thanks

File: web/web/api/v1/meter.py (hunk 3 above)
Comment (avpeery) [reply web/web/api/v1/meter.py:117]: My assumption is that is_active will always be set to the value False if True is not selected

File: web/web/api/v1/meter.py (hunk 5 above)
Comment (avpeery) [reply web/web/api/v1/meter.py:145]: Will do.

File: web/web/api/v1/meter.py (hunk 3 above)
Comment (jongoncalves) [reply web/web/api/v1/meter.py:117]: you mean it'll always be set to False on the client? Meaning the web client?
However, if I interact with this endpoint via the cmd line or postman, I won't know about this default. We should handle all of these types of assumptions on the server (backend) layer. We can never guarantee what type of bad data a client may be sending.

**Comment by jongoncalves** [comment]
This is starting to look real awesome! 

In the next PR we'll need to convert the api into a Flask blueprint, which would allow us remove that import statement from the `__init__.py`

File: web/web/api/v1/__init__.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+from web import app
+#enables trailing and non-trailing slash routes
+app.url_map.strict_slashes = False
```
Comment (jongoncalves) [inline web/web/api/v1/__init__.py:3]: let's put this on the `web/web/__init__.py` instead. That way if/when we create a public api or an api v2 we won't have to worry about remembering to set this again.

Comment (avpeery) [reply web/web/api/v1/__init__.py:3]: Ok will do

**Comment by jongoncalves** [comment]
@avpeery what other items are pending for this PR to complete?
I'm in between tasks right now so I can jump in to help move this through to master.

**Comment by avpeery** [comment]
Hi @jongoncalves Thanks for checking in.  There are just two things left on this PR to be done - usage of the ARW in place of the errors - however if you want to connect to make sure I am implementing it correctly that would be helpful! And using marshmallow dump for the GET specific meter id info route - I am stuck on brainstorming how to go about the marshmallow model for this since it isn't just a direct dump/load using an auto schema, like the POST or PUT routes.

**Comment by jongoncalves** [comment]
got it. I'm gonna pull your branch and checkout a few things

**Comment by avpeery** [comment]
ok I have some commit I haven't pushed to the branch yet, will do right now

**Pull request merged**